Renaming on declaration whose name is a String Literal

NamHaiBui opened the pull request:
<!--
Thank you for submitting a pull request!

Please verify that:
* [x] There is an associated issue in the `Backlog` milestone (**required**)
* [x] Code is up-to-date with the `main` branch
* [x] You've successfully run `hereby runtests` locally
* [x] There are new or updated unit tests validating the change

Refer to CONTRIBUTING.MD for more details.
  https://github.com/Microsoft/TypeScript/blob/main/CONTRIBUTING.md

** Please don't send typo fixes! **
Please don't send a PR solely for the purpose of fixing a typo, unless that
typo truly hurts understanding of the text. Each PR represents work for the
maintainers, and that work should provide commensurate value.

If you're interested in sending a PR, the issue tracker has many issues marked `help wanted`.
-->

Fixes #57026

NamHaiBui commented:
@jakebailey @DanielRosenwasser
Hi, can you review my pr if it's not too much trouble?
I am working on contributing to Typescript as my senior project.

jakebailey commented on tests/cases/fourslash/renameStringLiteralTypes5.ts:
```diff
@@ -1,11 +1,13 @@
+////declare const fnc_1: <K extends keyof Case_1>(p: K) => void
 ////
-////fn("Prop 1"/**/)
+////fnc_1("Att 2")
```
Why are these tests being modified rather than being added to, if anything?

NamHaiBui replied:
It was a duplicate test to renameStringLiteralTypes4 so I modified it to reflect one of the cases mentioned in the original issue.

jakebailey replied:
They are not duplicate tests. One is a `type`, the other is an `interface`.

NamHaiBui replied:
I'm sorry about that, I will revert changes and add an additional tests for `type` as well. Thank you and sorry for the oversight.

NamHaiBui commented:
@jakebailey 
I have updated the code and added in tests to check for both Type declaration and interface declaration.
Again, thank you so much for reviewing this and helping me.

NamHaiBui commented:
@jakebailey Hi, is it possible to get a review on this soon?
It would be greatly appreciated.
